docs/swarm/networking.md

```diff
@@ -0,0 +1,291 @@
+tasks for the service.
+
+Run `docker service ps <SERVICE>` to view the nodes where tasks are running for the service:
+
+```bash
+# From a manager node:
+$ docker service ps my_web
+
+ID                         NAME      IMAGE  NODE   DESIRED STATE  CURRENT STATE               ERROR
+63s86gf6a0ms34mvboniev7bs  my_web.1  nginx  node1  Running        Running 58 seconds ago
+6b3q2qbjveo4zauc6xig7au10  my_web.2  nginx  node2  Running        Running 58 seconds ago
+66u2hcrz0miqpc8h0y0f3v7aw  my_web.3  nginx  node3  Running        Running about a minute ago
+```
+
+![service vip image](images/service-vip.png)
+
+You can inspect the network from any node with a `Running` task for a service
+attached to the network:
+
+```bash
+docker network inspect <NETWORK>
+```
+
+The network information includes a list of the containers on the node that are
+attached to the network. For instance:
+
+```bash
+$ docker network inspect my-network
+[
+    {
+        "Name": "my-network",
+        "Id": "7m2rjx0a97n88wzr4nu8772r3",
+        "Scope": "swarm",
+        "Driver": "overlay",
+        "EnableIPv6": false,
+        "IPAM": {
+            "Driver": "default",
+            "Options": null,
+            "Config": [
+                {
+                    "Subnet": "10.0.9.0/24",
+                    "Gateway": "10.0.9.1"
+                }
+            ]
+        },
+        "Internal": false,
+        "Containers": {
+            "404d1dec939a021678132a35259c3604b9657649437e59060621a17edae7a819": {
+                "Name": "my_web.1.63s86gf6a0ms34mvboniev7bs",
+                "EndpointID": "3c9588d04db9bc2bf8749cb079689a3072c44c68e544944cbea8e4bc20eb7de7",
+                "MacAddress": "02:42:0a:00:09:03",
+                "IPv4Address": "10.0.9.3/24",
+                "IPv6Address": ""
+            }
+        },
+        "Options": {
+            "com.docker.network.driver.overlay.vxlanid_list": "257"
+        },
+        "Labels": {}
+    }
+]
+```
+
+In the example above, the container `my_web.1.63s86gf6a0ms34mvboniev7bs` for the
+`my_web` service is attached to the `my-network` network on node2.
+
+## Use swarm mode service discovery
+
+By default, when you create a service attached to a network, the swarm assigns
+the service a virtual IP address. You can inspect the service to view the
+virtual IP. For example:
+
+```bash
+$ docker service inspect \
+  --format=='{{json .Endpoint.VirtualIPs}}' \
+  my_web
+
+=[{"NetworkID":"7m2rjx0a97n88wzr4nu8772r3" "Addr":"10.0.0.2/24"}]
+```
+
+The service VIP means you don't need to expose service-specific ports to make
+the service available to other services on the same overlay network. The swarm's
+internal load balancer automatically distributes requests to the service VIP
+among the active tasks.
+
+For example, you can add a `busybox` service on the same network as the `nginx`
+service. The busybox service can access `nginx` using the DNS name `my_web`:
```

Comment:
it may be good to replace the underscores in our examples with dashes; Some software doesn't like that in DNS names, so better not use it in our examples (change to `my-web`, `my-busybox`)